### PR TITLE
Use min(size(A)...) instead of max(size(A)...) when computing the numrical rank

### LINKED ISSUE #461: Use min(m,n) instead of max(m,n) when determining numerical rank in pinv and \
Following up on https://github.com/JuliaLang/julia/pull/23067#discussion_r131481492. I did some simulations and it seems that error is determined by the minimum and not the maximum size. This would also make the vector and matrix case consistent.

### LINKED ISSUE #23386: Use min(m,n) instead of max(m,n) when determining numerical rank in pinv and \
Following up on https://github.com/JuliaLang/julia/pull/23067#discussion_r131481492. I did some simulations and it seems that error is determined by the minimum and not the maximum size. This would also make the vector and matrix case consistent.

### PR DESCRIPTION
**Pull request by andreasnoack**
Pivoted Cholesky still uses the LAPACK default. Since that factorization doesn't use orthogonal transformations, the tolerance should most likely be different so better to leave it to LAPACK. We can revisit when we get a generic pivoted Cholesky.

Fixes JuliaLang/LinearAlgebra.jl#461

### CODE REVIEW

File: base/linalg/qr.jl [hunk 1]
```diff
@@ -194,7 +194,7 @@ function qrfactPivotedUnblocked!(A::StridedMatrix)
 end
 
 # LAPACK version
-qrfact!(A::StridedMatrix{<:BlasFloat}, ::Val{false}) = QRCompactWY(LAPACK.geqrt!(A, min(minimum(size(A)), 36))...)
+qrfact!(A::StridedMatrix{<:BlasFloat}, ::Val{false}) = QRCompactWY(LAPACK.geqrt!(A, min(min(size(A)...), 36))...)
```
Comment (maleadt) [inline base/linalg/qr.jl]: Out of curiosity, why this change? `minimum` seems faster than `min` + splat (with some superficial benchmarking)

Comment (StefanKarpinski) [reply base/linalg/qr.jl]: Generally, yes, but `size(A)` is a tuple whose size type inference generally knows and specializes code on. Of course, `minimum` may be specialized for tuples as well...

Comment (StefanKarpinski) [reply base/linalg/qr.jl]: Nope, just checked:
```jl
julia> @code_llvm min((1,2)...)

; Function min
; Location: promotion.jl
define i64 @jlsys_min_44934(i64, i64) #0 {
top:
; Location: promotion.jl:388
  %2 = icmp sge i64 %1, %0
  %3 = select i1 %2, i64 %0, i64 %1
  ret i64 %3
}

julia> @code_llvm minimum((1,2))

; Function minimum
; Location: reduce.jl
define i64 @julia_minimum_61878([2 x i64] addrspace(11)* nocapture readonly dereferenceable(16)) #0 {
pass:
  %gcframe1 = alloca [3 x %jl_value_t addrspace(10)*], align 8
  %gcframe1.sub = getelementptr inbounds [3 x %jl_value_t addrspace(10)*], [3 x %jl_value_t addrspace(10)*]* %gcframe1, i64 0, i64 0
  %1 = getelementptr inbounds [3 x %jl_value_t addrspace(10)*], [3 x %jl_value_t addrspace(10)*]* %gcframe1, i64 0, i64 1
  %2 = bitcast %jl_value_t addrspace(10)** %1 to i8*
  call void @llvm.memset.p0i8.i32(i8* %2, i8 0, i32 16, i32 8, i1 false)
  %3 = call %jl_value_t*** @jl_get_ptls_states() #5
  %4 = bitcast [3 x %jl_value_t addrspace(10)*]* %gcframe1 to i64*
  store i64 2, i64* %4, align 8
  %5 = bitcast %jl_value_t*** %3 to i64*
  %6 = load i64, i64* %5, align 8
  %7 = getelementptr [3 x %jl_value_t addrspace(10)*], [3 x %jl_value_t addrspace(10)*]* %gcframe1, i64 0, i64 1
  %8 = bitcast %jl_value_t addrspace(10)** %7 to i64*
  store i64 %6, i64* %8, align 8
  %9 = bitcast %jl_value_t*** %3 to %jl_value_t addrspace(10)***
  store %jl_value_t addrspace(10)** %gcframe1.sub, %jl_value_t addrspace(10)*** %9, align 8
; Location: reduce.jl:471
; Function mapreduce; {
; Location: reduce.jl:209
; Function mapfoldl; {
; Location: reduce.jl:71
  %10 = getelementptr [2 x i64], [2 x i64] addrspace(11)* %0, i64 0, i64 0
; Location: reduce.jl:73
  %11 = load i64, i64 addrspace(11)* %10, align 8
  %12 = call i64 @julia_mapfoldl_impl_61879(i64 %11, [2 x i64] addrspace(11)* nocapture readonly %0, i64 2)
;}}
  %13 = load i64, i64* %8, align 8
  store i64 %13, i64* %5, align 8
  ret i64 %12
}
```

Comment (maleadt) [reply base/linalg/qr.jl]: I see, thanks!

**Review by fredrikekre — APPROVED** [review]
